.github/workflows/ci.yaml

```diff
@@ -15,20 +19,136 @@ on:
       - cryostat-v[0-9]+.[0-9]+
 
 jobs:
-  build:
+  build-core:
+    runs-on: ubuntu-latest
+    outputs:
+      test_version: ${{ steps.get_core_version.outputs.test_version }}
+    steps:
+    - uses: actions/checkout@v4
+    - id: get_core_version
+      run: |
+        echo "test_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_OUTPUT
+        echo "name=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)" >> $GITHUB_OUTPUT
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: 'maven'
+    - run: mvn -B -U clean install
+    - uses: actions/upload-artifact@v3
```

Comment:
See: https://github.com/skjolber/maven-cache-github-action#cache-scopes

According to this, the cache should not conflict between concurrent workflow (i.e. multiple PRs). I think we can just restore the cache in image build and remove this.